### ISSUE TITLE
[4.0.0] indent with flatTernayExpressions and consequent on new line

### ISSUE THREAD

**Issue by zaygraveyard** [eslint/eslint#8712]
**Tell us about your environment**

* **ESLint Version:** v4.0.0
* **Node Version:** v8.1.0
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```json
{
  "rules": {
    "indent": ["error", 2, {"flatTernaryExpressions": true}]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
Math.random()
? true
: false
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
<text>
  2:1  error  Expected indentation of 2 spaces but found 0  indent
  3:1  error  Expected indentation of 2 spaces but found 0  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```

**Comment by mysticatea** [comment]
Thank you for this issue.

However, I think that it works as intended.
The `flatTernaryExpressions` is introduced to support the following style. In other cases, `indent` rule reports the same indentation as normal.

```js
var result = 
    a > 100 ? 100 :
    a < 0   ? 0 :
    /* else */ a ;
```

See also: https://github.com/eslint/eslint/pull/7618#issuecomment-268943294

**Comment by zaygraveyard** [comment]
Can an option be added to support this style?
Maybe something like `ConditionalExpression: 0`?
I'm willing to work on it if needed.
Thanks

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
